Fix tests/travis/nbexecuter.py to work under IPython 3

mattpap opened the pull request:
This updates notebooks to nbformat v4 (current) as well.

fixes: #2017

mattpap commented:
You can test this locally with: `pytest tests/travis/test_examples.py nbex`.

bryevdv commented:
looks like one failure:
```
======================================================================
ERROR: test_nbexecuter (travis.test_examples.TestExamples)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/bokeh/bokeh/tests/travis/test_examples.py", line 24, in test_nbexecuter
    import nbexecuter
ImportError: No module named nbexecuter
```

mattpap commented:
I replaced the last commit with, hopefully, a correct one. Apparently it was only pytest that had a problem with relative imports. I changed the nbexecuter import to be relative, so that it works in all pythons.
